bionlp/taggers/rnn_feature/tagger_utils.py

```diff
@@ -123,6 +123,41 @@ def trim_tags(tagged_data):
                           for x, y in tagged_data[i]]
     return tagged_data
 
+def get_regular_or_capitalized_embeddings(w2i, i2w, mdl, params):
+    embedding_vector_size = params['emb1_size']
+    embeddings = []
+    for i in range(len(w2i)):
+        term_emb = None
+        if i in i2w:
+            regular_term = str(i2w[i])
+            capitalized_term = regular_term.capitalize()
+            if regular_term in mdl:
+                term_emb = mdl[regular_term]
+            elif capitalized_term in mdl:
+                term_emb = mdl[capitalized_term]
+
```

Comment:
There's two cases that can lead to `term_emb` being initialized as a zero vector:

- `i` not being one of the keys in `i2w`
- neither the regular nor the capitalized writing of the term being one of the keys of `mdl`

I initially set `term_emb` to `None`, and it is initialized correctly when a corresponding word embedding vector is found in `mdl`. The `else` case would simply set it to `None` once more.